rca/api_content/content.py

```diff
@@ -150,9 +156,8 @@ def pull_news_and_events(programme_type_slug=None):
     news_items_to_get = 3
 
     events_data = []
-    response = try_and_fetch(events_url)
-    data = response.json()
-    if data["meta"]["total_count"] > 0:
+    data = fetch_data(events_url)
+    if data and data["meta"]["total_count"] > 0:
```

Comment:
Potential for index errors here?

Reply:
> Does this code run on page load or as part of a management command?

Runs on page load, cache is regenerated after 15 mins though